std/haxe/macro/Expr.hx

```diff
@@ -169,154 +213,603 @@ enum Unop {
+
+	/**
+		The position of the metadata entry.
+	**/
 	pos : Position
 }
 
+/**
+	Represents metadata for the AST.
+**/
 typedef Metadata = Array<MetadataEntry>;
 
+/**
+	Represents a field for the AST.
+**/
 typedef Field = {
+	/**
+		The name of the field.
+	**/
 	var name : String;
+
+	/**
+		The documentation of the field. If the field has no documentation, the 
+		value is `null`.
+	**/
 	@:optional var doc : Null<String>;
+
+	/**
+		The access modifiers of the field. By default fields have private access.
+		@see http://haxe.org/manual/class-field-access-modifier.html
+	**/
 	@:optional var access : Array<Access>;
+
+	/**
+		The type of the field.
+	**/
 	var kind : FieldType;
+
+	/**
+		The position of the field.
+	**/
 	var pos : Position;
+
+	/**
+		The optional metadata of the field.
+	**/
 	@:optional var meta : Metadata;
 }
 
+/**
+	Represents an access modifier.
+	@see http://haxe.org/manual/class-field-access-modifier.html
+**/
 enum Access {
+
+	/**
+		Public access modifier, grants access from anywhere.
+		@see http://haxe.org/manual/class-field-visibility.html
+	**/
 	APublic;
+
+	/**
+		Private access modifier, grants access to class and its sub-classes 
+		only.
+		@see http://haxe.org/manual/class-field-visibility.html
+	**/
 	APrivate;
+
+	/**
+		Static access modifier. 
+	**/
 	AStatic;
+
+	/**
+		Override access modifier, required when a field is declared which also 
+		exists on a parent class. This modifier is only allowed on 
+		method fields.
+	**/
 	AOverride;
+
+	/**
+		Dynamic (re-)bindable access modifier. 
+		@see http://haxe.org/manual/class-field-dynamic.html
+	**/
 	ADynamic;
+
+	/**
+		Inline access modifier. Allows expressions to be directly inserted in 
+		place of calls to them.
+	**/
 	AInline;
+
+	/**
+		Macros access modifier. Allows expression macro functions. These are 
+		normal functions which are executed as soon as they are typed.
+	**/
 	AMacro;
 }
 
+/**
+	Represents the field type for the AST.
+**/
 enum FieldType {
+	/**
+		Represents a variable structure.
+	**/
 	FVar( t : Null<ComplexType>, ?e : Null<Expr> );
+
+	/**
+		Represents a function structure.
+	**/
 	FFun( f : Function );
+
+	/**
+		Represents a property with getter and setter structure.
+	**/
 	FProp( get : String, set : String, ?t : Null<ComplexType>, ?e : Null<Expr> );
 }
 
+/**
+	Represents a type definition for the AST.
+**/
 typedef TypeDefinition = {
+	/**
+		The package of the type definition.
+	**/
 	var pack : Array<String>;
+
+	/**
+		The name of the type definition.
+	**/
 	var name : String;
+
+	/**
+		The position to the type definition.
+	**/
 	var pos : Position;
+
+	/**
+		The metadata of the type definition.
+	**/
 	@:optional var meta : Metadata;
+
+	/**
+		The paramater type declarations of the type definition.
+	**/
 	@:optional var params : Array<TypeParamDecl>;
+
+	/**
+		Whether or not the type is extern.
+	**/
 	@:optional var isExtern : Bool;
+
+	/**
+		The type kind of the type definition.
```

Comment:
:smile: They all feel like _"The structure of the type kind type typed as the type kind definition structure definitions."_